fastlane/Fastfile

```diff
@@ -71,28 +80,61 @@ lane :distribute_release_build do |_options|
   release_tag = get_required_env('BUILDKITE_TAG')
   builds = distribute_builds(release_tag:)
 
-  slack(
-    username: 'CI Bot',
-    icon_url: 'https://octodex.github.com/images/jenktocat.jpg',
-    message: ":tada: #{release_tag} is now available! :tada:",
-    channel: '#dotcom-studio',
-    success: true,
-    slack_url: get_required_env('SLACK_WEBHOOK'),
-    payload: {
-      ':studio-app-icon-mac: Release available for': builds.each_value.map { |b| "[#{b[:name]}](#{b[:cdn_url]})" }.join(', ')
-    },
-    default_payloads: []
+  unless DRY_RUN
+    slack(
+      username: 'CI Bot',
+      icon_url: 'https://octodex.github.com/images/jenktocat.jpg',
+      message: ":tada: #{release_tag} is now available! :tada:",
+      channel: '#dotcom-studio',
+      success: true,
+      slack_url: get_required_env('SLACK_WEBHOOK'),
+      payload: {
+        ':studio-app-icon-mac: Release available for': builds.each_value.map { |b| "[#{b[:name]}](#{b[:cdn_url]})" }.join(', ')
+      },
+      default_payloads: []
+    )
+  end
+end
+
+def upload_to_s3_with_dry_run(bucket:, key:, file:, if_exists:, auto_prefix:)
+  if DRY_RUN
+    UI.message("[DRY RUN] Would upload file: #{file}")
+    UI.message("           to S3 bucket: #{bucket}")
+    UI.message("           with key: #{key}")
+    UI.message("           if_exists: #{if_exists}")
+    return true
+  end
+
+  upload_to_s3(
+    bucket: bucket,
+    key: key,
+    file: file,
+    if_exists: if_exists,
+    auto_prefix: auto_prefix
   )
 end
 
+def clear_cloudfront_cache_with_dry_run(paths:, commit_hash:)
+  if DRY_RUN
+    UI.message("[DRY RUN] Would invalidate CloudFront cache:")
+    UI.message("           Distribution: EF4A01YASGPY5")
```

Comment:
Same question as `upload_to_s3` vs `upload_to_s3_with_dry_run` here.

As a bonus, it would remove the duplication of the `EF4A01YASGPY5` value across two methods.